include/sys/device_mmio.h

```diff
@@ -218,27 +218,27 @@ struct z_device_mmio_rom {
 	((struct z_device_mmio_rom *)((device)->config))
 
 /**
- * @def DEVICE_MMIO_ROM_INIT(instance)
+ * @def DEVICE_MMIO_ROM_INIT(node_id)
  *
  * @brief Initialize a DEVICE_MMIO_ROM member
  *
  * Initialize MMIO-related information within a specific instance of
  * a device config struct, using information from DTS.
  *
- * Example for an instance of a driver belonging to the "foo" subsystem:
+ * Example for a driver belonging to the "foo" subsystem:
  *
  * struct foo_config my_config = {
- *	DEVICE_MMIO_ROM_INIT(instance),
+ *	DEVICE_MMIO_ROM_INIT(DT_DRV_INST(...)),
  *	.baz = 2;
  *	...
  * }
  *
  * @see DEVICE_MMIO_ROM()
  *
- * @param instance DTS instance
+ * @param node_id DTS node_id
  */
-#define DEVICE_MMIO_ROM_INIT(instance) \
-	._mmio = Z_DEVICE_MMIO_ROM_INITIALIZER(instance)
+#define DEVICE_MMIO_ROM_INIT(node_id) \
```

Comment:
I guess you decided against adding a seaprate DEVICE_MMIO_ROM_INST_INIT(inst) to avoid the DT_DRV_INST boilerplate in the callers? No objections if you want to keep the API surface here smaller, but I had suggested that in the original issue to keep the typing down. Your call, of course.

Reply:
this file already has a lot of different macro variants in it, and I felt this would only save user typing a few characters.